File: survey/src/main/java/com/onboarding/survey/service/SurveyService.java
package com.onboarding.survey.service;

import com.onboarding.core.global.exception.CustomException;
import com.onboarding.core.global.exception.enums.ErrorCode;
import com.onboarding.survey.entity.Question;
import com.onboarding.survey.entity.Survey;
import com.onboarding.survey.repository.QuestionRepository;
import com.onboarding.survey.repository.SurveyRepository;
import java.util.List;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Slf4j
@Service
@RequiredArgsConstructor
public class SurveyService {

  private final SurveyRepository surveyRepository;
  private final QuestionRepository questionRepository;

  public void createSurvey(Survey survey) {
    surveyRepository.save(survey);
  }

  public Survey getSurveyById(Long surveyId) {
    return surveyRepository.findByIdWithActiveQuestions(surveyId).orElseThrow(() -> new CustomException(
        "Required question is missing: ", ErrorCode.ENTITY_NOT_FOUND));
  }

  public Optional<Survey> findSurveyById(Long surveyId) {
    return surveyRepository.findByIdWithActiveQuestions(surveyId);
  }

  public List<Question> findQuestionsBySurveyId(Long surveyId) {
    return questionRepository.findBySurveyId(surveyId);
  }
}

File: survey-api/src/main/kotlin/com/ic/surveyapi/service/SurveyService.kt
package com.ic.surveyapi.service

import com.ic.surveydata.entity.SurveyForm
import com.ic.surveydata.service.SurveyFormDataHandler
import org.springframework.stereotype.Service

@Service
class SurveyService(
    private val surveyFormDataHandler: SurveyFormDataHandler,
) {
    fun createSurvey(surveyForm: SurveyForm): SurveyForm {
        surveyFormDataHandler.createForm(surveyForm = surveyForm)
//       surveyRepository.save(survey)
        return SurveyForm()
    }
}
